src/test/java/fi/aalto/cs/apluscourses/presentation/base/BaseListViewModelTest.java

```diff
@@ -0,0 +1,96 @@
+package fi.aalto.cs.apluscourses.presentation.base;
+
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class BaseListViewModelTest {
+
+  @Test
+  public void testCreateBaseListViewModel() {
+    Object model1 = new Object();
+    Object model2 = new Object();
+    Object model3 = new Object();
+
+    List<Object> models = new ArrayList<>();
+    models.add(model1);
+    models.add(model2);
+    models.add(model3);
+
+    ListElementViewModel<?> listElementViewModel1 = new ListElementViewModel<>(model1);
+    ListElementViewModel<?> listElementViewModel2 = new ListElementViewModel<>(model2);
+    ListElementViewModel<?> listElementViewModel3 = new ListElementViewModel<>(model3);
+
+    Map<Object, ListElementViewModel<?>> elementMap = new HashMap<>();
```

Comment:
there is this clause when describing a HashMap:

> [This class makes no guarantees as to the order of the map; in particular, it does not guarantee that the order will remain constant over time.](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html)

could this "no guarantee" have an effect here, when accessing elements in assertions?

Reply:
That map is not used in assertions. It is only used mapping model objects to view model objects in `BaseListViewModel` constructor.